conn.go

```diff
@@ -36,13 +36,21 @@ type clientConn struct {
 	wg         sync.WaitGroup
 	sync.Mutex                          // protects inflight
 	inflight   map[uint32]chan<- result // outstanding requests
-	errCh      chan error
+
+	errCond *sync.Cond
+	err     error
 }
 
 // Wait blocks until the conn has shut down, and return the error
-// causing the shutdown.
+// causing the shutdown. It can be called concurrently from multiple
+// goroutines.
 func (c *clientConn) Wait() error {
-	return <-c.errCh
+	c.errCond.L.Lock()
+	defer c.errCond.L.Unlock()
+	for c.err == nil {
```

Comment:
Now that I think about it, this `Wait` should probably return if the connection is shut down properly with a `Close` as well, right? In that case, we would want `c.err == nil`.